frontend/packages/operator-lifecycle-manager/src/components/modals/uninstall-operator-modal.tsx

```diff
@@ -88,28 +170,51 @@ export const UninstallOperatorModal: React.FC<UninstallOperatorModalProps> = ({
               },
               {},
               deleteOptions,
-            ).catch(() => Promise.resolve()),
+            ),
           ]
         : []),
       ...(removePlugins
         ? [k8sPatch(ConsoleOperatorConfigModel, consoleOperatorConfig, patch)]
         : []),
     ];
+    const operandDeletionPromises = isDeleteAllOperandsSelected
+      ? operands.map((operand: K8sResourceCommon) => {
+          const model = modelFor(referenceFor(operand));
+          return k8sKill(model, operand, {}, deleteOptions);
+        })
+      : [];
+    if (operandDeletionPromises.length > 0) {
+      // The promise returned by Promise.allSettled() never rejects; no need for catch-or-return.
+      // eslint-disable-next-line promise/catch-or-return
+      handleOperandsDeletePromise(operandDeletionPromises).then(
+        (results: PromiseSettledResult<any>[]) => {
+          const operandsWithErrors: OperandWithErrorProps[] = results.reduce((acc, curr, i) => {
+            return curr.status === 'rejected'
+              ? acc.concat({ operand: operands[i], errorMessage: curr.reason })
+              : acc;
+          }, [] as OperandWithErrorProps[]);
+          setOperandDeletionErrors(operandsWithErrors);
+          setOperandsDeleteFinished(true);
+        },
+      );
+    }
+    setOperandsDeleteFinished(!isDeleteAllOperandsSelected);
 
-    handlePromise(Promise.all(allPromises))
+    // TODO .finally causes "ESLint: Expected catch() or return (promise/catch-or-return)" !?
+    handleOperatorUninstallPromise(Promise.all(operatorUninstallPromises))
       .then(() => {
-        close();
-        if (
-          window.location.pathname.split('/').includes(subscription.metadata.name) ||
-          window.location.pathname.split('/').includes(subscription.status.installedCSV)
-        ) {
-          history.push(resourceListPathFromModel(ClusterServiceVersionModel, getActiveNamespace()));
-        }
+        setOperatorUninstallFinished(true);
       })
-      .catch(() => {});
+      .catch(() => {
+        setOperatorUninstallFinished(true);
+      });
+    /* .finally(() => {
```

Comment:
should we remove the `catch` block and leave just `finally` block ?

Reply:
Hi @jhadvig, pls see the comment in the code: `// TODO .finally causes "ESLint: Expected catch() or return (promise/catch-or-return)" !?`  For some reason, I'm having trouble implementing the `finally` here?

Reply:
Tried `.finally()` again, but it doesn't get hit, maybe due to overriding `.finally()` in [usePromiseHandler](https://github.com/openshift/console/pull/9142/files#diff-2cfda67e7f418182a003111b61367ce0fd9f0a0908dc5c196b3bb6ce9e7a5171R17)?

Reply:
Further testing revealed we do need a `catch` here as `usePromiseHandler` returns a promise reject, so we need to catch it here in the calling `handleOperatorUninstallPromise`.